src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -34,46 +36,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Boolean to update User's main page 
     Boolean showNewTab = false;
 
+    User user;
     if (userService.isUserLoggedIn()) {
-      userEmail = userService.getCurrentUser().getEmail();
       userId = userService.getCurrentUser().getUserId();
+      userEmail = userService.getCurrentUser().getEmail();
       logoutUrl = userService.createLogoutURL(urlToRedirect);
       showNewTab = true;
     } else {
       loginUrl = userService.createLoginURL(urlToRedirect);
     }
 
-    User user = new User(userId, loginUrl, logoutUrl, userEmail, showNewTab);
-    
-    if (userId != "null") {
-        checkUserInDatastore(user);
+    user = new User(userId, userEmail);
+
+    if ((userId != "null") && (!isUserInDatastore(user))) {
```

Comment:
We don't need these extra parens — it's fairly clear to most Java programmers that != binds less tightly than && (otherwise common conditionals like `if (a == 1 && b == 2)` would not work).